packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+import {
+  CommittedTransactionInfo,
+  GatewayApiClient,
+  GatewayStatusResponse,
+  StateEntityDetailsRequest,
+  StateEntityDetailsResponse,
+  TransactionCommittedDetailsRequest,
+  TransactionCommittedDetailsResponse,
+  TransactionPreviewRequest,
+  TransactionPreviewResponse,
+  TransactionSubmitResponse,
+} from '@radixdlt/babylon-gateway-api-sdk'
+import {
+  Convert,
+  Curve,
+  LTSRadixEngineToolkit,
+  NetworkId,
+  PrivateKey,
+  RadixEngineToolkit,
+  TransactionBuilder,
+  TransactionManifest,
+  generateRandomNonce,
+} from '@radixdlt/radix-engine-toolkit'
+import {
+  AssetInfo,
+  Balance,
+  BaseXChainClient,
+  FeeOption,
+  FeeType,
+  Fees,
+  Network,
+  PreparedTx,
+  Tx,
+  TxHistoryParams,
+  TxParams,
+  TxType,
+  TxsPage,
+  XChainClientParams,
+} from '@xchainjs/xchain-client'
+import { getSeed } from '@xchainjs/xchain-crypto/lib'
+import { Address, Asset, assetAmount, assetToBase, baseAmount } from '@xchainjs/xchain-util'
+import BIP32Factory, { BIP32Interface } from 'bip32'
+import { derivePath } from 'ed25519-hd-key'
+import * as ecc from 'tiny-secp256k1'
+// eslint-disable-next-line ordered-imports/ordered-imports
+import {
+  AssetXRD,
+  MAINNET_GATEWAY_URL,
+  RadixChain,
+  STOKENET_GATEWAY_URL,
+  TransferTransactionManifest,
+  XRD_DECIMAL,
+  xrdRootDerivationPaths,
+} from './const'
+
+/**
+ * Custom Radix client
+ */
+
+export default class Client extends BaseXChainClient {
+  gatewayApiClient: GatewayApiClient
+  curve: Curve
+  constructor(params: XChainClientParams, curve: Curve) {
+    super(RadixChain, {
+      network: params.network,
+      phrase: params.phrase,
+      rootDerivationPaths: xrdRootDerivationPaths,
+      feeBounds: params.feeBounds,
+    })
+    this.curve = curve
+    this.gatewayApiClient = GatewayApiClient.initialize({
+      networkId: this.getRadixNetwork(),
+      applicationName: 'xchainjs',
+    })
+  }
+
+  /**
+   * Get an estimated fee for a test transaction that involves sending
+   * XRD from one account to another
+   *
+   * @returns {Fee} An estimated fee
+   */
+  async getFees(): Promise<Fees> {
+    try {
+      const gatewayStatusResponse: GatewayStatusResponse = await this.gatewayApiClient.status.getCurrent()
+      const transactionPreviewRequest: TransactionPreviewRequest = {
+        manifest: TransferTransactionManifest,
+        start_epoch_inclusive: gatewayStatusResponse.ledger_state.epoch,
+        end_epoch_exclusive: gatewayStatusResponse.ledger_state.epoch + 10,
+        tip_percentage: 10,
+        nonce: Math.floor(Math.random() * 1000000),
+        signer_public_keys: [
```

Comment:
Why are we using `signer_public_keys` and not using `notary_is_signatory = true` and `notary_public_key`? This preview tx should be identical to the transfer tx we're submitting. In fact, I would even recommend using the transaction created from `transfer` in here.

Reply:
@0xOmarA I guess you meant using the transaction from `prepareTx` not from `transfer`. TransactionPreviewRequest needs a manifest

Reply:
I have now updated getFees to get the manifest from prepareTx. 
btw I was using signer_public_keys because it is mandatory in the sdk

Reply:
> @0xOmarA I guess you meant using the transaction from prepareTx not from transfer. TransactionPreviewRequest needs a manifest

Either would be fine, since a transaction intent contains a manifest. But we're more interested in the intent because ideally we would like to do a preview using the same header as the intent.

> I have now updated getFees to get the manifest from prepareTx.
btw I was using signer_public_keys because it is mandatory in the sdk

On the `signer_public_keys` being required, we can just set it to be an empty array.

Reply:
ok I'm now using the output from prepareTx and passing an empty list to `signer_public_keys`